### ISSUE TITLE
Problem with path names containing spaces

### ISSUE THREAD

**Issue by mbertini** [johnthagen/clion-cppcheck#13]
### Environment

* Operating System (e.g. Ubuntu 16.04 x64): macOS 10.12.6
* IDE Version (e.g. CLion 2016.3.2): CLion 2017.3
* Cppcheck executable version (`cppcheck --version`): 1.81
* Cppcheck plugin version: 1.0.10
* Exact strings used in cppcheck plugin options:
  * **cppcheck path**: /usr/local/bin/cppcheck
  * **cppcheck options**: --enable=all,performance,portability,style --language=c++ --force

### Expected behaviour
Plugin should report the cppcheck warnings even if the path name of the project contains a space
### Actual behaviour
If the path name of the project contains a space, e.g. "/User/myname/Workspace/path with space/projectName" then no error/warnings from cppcheck are shown. Running cppcheck in the CLIon terminal correctly shows them, i..e. cppcheck has no issues with path names containing spaces
### Steps to reproduce the behaviour
create a project in a path name containing paces and some cppcheck error, see that no error/warning is shown. Move the project to a path name without spaces and reopen CLion, then recompile. Errors are shown.

**Comment by johnthagen** [comment]
Hmm, can't reproduce on Windows. What's interesting is the plugin does already try to quote the file path so that spaces shouldn't matter.

https://github.com/johnthagen/clion-cppcheck/blob/master/src/com/github/johnthagen/cppcheck/CppcheckInspection.java#L144

**Comment by johnthagen** [comment]
> Running cppcheck in the CLIon terminal correctly shows them, i..e. cppcheck has no issues with path names containing spaces

Is the terminal open to the `projectName` folder already? In that case, you might be passing relative paths so the spacing wouldn't really be considered

**Comment by johnthagen** [comment]
Confirmed problem exists on macOS.

**Comment by johnthagen** [comment]
In this situation Cppcheck is returning 1 as the exit status (error) and not writing results to stderr as it normally would.

**Comment by johnthagen** [comment]
What I'm guessing is happening is that 

```
final Process process = Runtime.getRuntime().exec(executionString);
```

Is stripping the `" "` off of the file path on macOS (but not on Windows). You get the same error if you run cppcheck like:

```
$ cppcheck /User/myname/Workspace/path with space/projectName/main.cpp
```

**Comment by fastasturtle** [comment]
@johnthagen see https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/execution/configurations/GeneralCommandLine.java

**Comment by johnthagen** [comment]
@mbertini When 1.1.0 is approved by Jetbrains, please test it with your path-space issues.

**Comment by johnthagen** [comment]
@mbertini [1.1.0](https://plugins.jetbrains.com/plugin/8143-cppcheck) has been released.

**Comment by mbertini** [comment]
I confirm the bug is solved. Thanks !

**Comment by johnthagen** [comment]
Great, @fastasturtle gets the credit for this one.
